### DISCUSSION TITLE
Message Viewer Quick Improvements

### DISCUSSION THREAD

**Discussion by jlrobins** [confluentinc/vscode#126] [Ideas]
### Goals

1. Quick message viewer improvement proposals.

### Non-Goals

1. Things individually accomplishable in a half day.

### Background

Message viewer is a fertile ground for short term improvements.

### Proposal

1. Reorder the 'WORKING VIEW SETTINGS' to be 'Consume mode' first, then 'Partitions'. Arguably choosing when to consume from is higher precedence / more likely to be considered than partition number.
2. ✅ ~~Allow Consume From: timestamp to accept an ISO formatted string in addition to epoch timestamp. We display 'em, should be able to eat 'em.~~
4. User preference section for message viewing, starting with preferred default timestamp format: Epoch, ISO GMT, ISO Local timezone. Would affect the timezone column, the 'N messages streamed since ...', and the histogram column tooltips (yes, showing those in epoch might be useful to some people -- will at least be consistent). Could be extended for arbitrary timezones, but GMT and local are the biggest cases.
5. ✅ ~~Column settings for timestamp format extended to include 'ISO Local Timezone', used to override the user setting above for all of the timestamp displays. Is then the 'page timestamp / timezone setting'.~~
5. Tooltip what the '{}' button does. Is a hidden feature!
6. Add in a couple more convenience options for consume mode like 'From Midnight Today' (timezone as interpreted per the page timezone setting. And also 'Past 5 Minutes', 'Past 15 Minutes', ....
7. New leftmost column in the message viewing table with a clickable cell allowing individual messages to be selected. Selected message rows will be highlighted as 'distinguished'. Clicking in the column header above this new column would then toggle between filtering by 'selected messages' or not.

**Comment by theseeker58** [comment]
Adding a list of custom headers could be pretty useful as well. It could be a hover pop-up box just like the message payload
